script/deploy/query-deployment-logs.ts

```diff
@@ -0,0 +1,638 @@
+  }
+
+  public async connect(): Promise<void> {
+    try {
+      await this.client.connect()
+      this.db = this.client.db(config.databaseName)
+      const collectionName = this.environment
+      this.collection = this.db.collection<IDeploymentRecord>(collectionName)
+      consola.info(`Connected to MongoDB collection: ${collectionName}`)
+    } catch (error) {
+      consola.error('Failed to connect to MongoDB:', error)
+      throw error
+    }
+  }
+
+  public async getLatestDeployment(
+    contractName: string,
+    network: string
+  ): Promise<IDeploymentRecord | null> {
+    return this.collection.findOne(
+      { contractName, network },
+      { sort: { timestamp: -1 } }
+    )
+  }
+
+  public async listDeployments(
+    contractName?: string,
+    network?: string,
+    limit = 50,
+    page = 1
+  ): Promise<{
+    data: IDeploymentRecord[]
+    pagination: {
+      page: number
+      limit: number
+      total: number
+      totalPages: number
+      hasNext: boolean
+      hasPrev: boolean
+    }
+  }> {
+    const filter: Record<string, unknown> = {}
+    if (contractName) filter.contractName = contractName
+    if (network) filter.network = network
+
+    const skip = (page - 1) * limit
+    const total = await this.collection.countDocuments(filter)
+    const totalPages = Math.ceil(total / limit)
+
+    const data = await this.collection
+      .find(filter)
+      .sort({ timestamp: -1 })
+      .skip(skip)
+      .limit(limit)
+      .toArray()
+
+    return {
+      data,
+      pagination: {
+        page,
+        limit,
+        total,
+        totalPages,
+        hasNext: page < totalPages,
+        hasPrev: page > 1,
+      },
+    }
+  }
+
+  public async findByAddress(
+    address: string
+  ): Promise<IDeploymentRecord | null> {
+    return this.collection.findOne({ address })
+  }
+
+  public async filterDeployments(filters: {
+    contractName?: string
+    network?: string
+    version?: string
+    verified?: boolean
+    limit?: number
+  }): Promise<IDeploymentRecord[]> {
+    const query: Record<string, unknown> = {}
+
+    if (filters.contractName) query.contractName = filters.contractName
+    if (filters.network) query.network = filters.network
+    if (filters.version) query.version = filters.version
+    if (filters.verified !== undefined) query.verified = filters.verified
+
+    return this.collection
+      .find(query)
+      .sort({ timestamp: -1 })
+      .limit(filters.limit || 50)
+      .toArray()
+  }
+
+  public async getDeploymentHistory(
+    contractName: string,
+    network: string
+  ): Promise<IDeploymentRecord[]> {
+    return this.collection
+      .find({ contractName, network })
+      .sort({ timestamp: -1 })
+      .toArray()
+  }
+
+  public async disconnect(): Promise<void> {
+    await this.client.close()
+  }
+}
+
+// Helper function to format deployment record for display
+function formatDeployment(deployment: IDeploymentRecord): string {
```

Comment:
I think it's worth to add createdAt and updatedAt then? wdyt? or name it as summary

Reply:
Fixed: 3298710023497f4a30595870fb38125c2f17787d